doc/Database.md

```diff
@@ -0,0 +1,105 @@
+
+
+### `accumulated_fee` table
+
+Note sure why we need this.
+
+| Field name | SQL Type | D type     | Attributes  | Comment             |
+|------------|----------|------------|-------------|---------------------|
+| public_key | TEXT     | PublicKey  | PRIMARY KEY |                     |
+| fee        | TEXT     | Enrollment |             | Should be `INTEGER` |
+
+## Cache DB
+
+### `tx_pool` table
+
+Stores the pending transaction. Should be made readable.
+
+| Field name | SQL Type | D type      | Attributes  | Comment           |
+|------------|----------|-------------|-------------|-------------------|
+| key        | TEXT     | Hash        | PRIMARY KEY | Transaction hash  |
+| val        | BLOB     | Transaction | NOT NULL    | Binary-serialized |
+
+### `enrollment_pool` table
+
+Store the `Enrollment` that haven't made it to a `Block` yet. Should be made readable.
+
+| Field name   | SQL Type | D type     | Attributes  | Comment                                                     |
+|--------------|----------|------------|-------------|-------------------------------------------------------------|
+| key          | TEXT     | Hash       | PRIMARY KEY | The `utxo` field, not the hash of the `Enrollment` itself   |
+| val          | BLOB     | Enrollment |             | Binary serialized `Enrollment`, should be converted to text |
+| avail_height | INTEGER  | Height     |             | It's not actually a hash but the key, should be `TEXT`      |
+
+
+## SCP envelope DB
+
+This DB is in its own file and should probably merged in one or the other.
+
+| Field name | SQL Type | D type      | Attributes                | Comment                             |
+|------------|----------|-------------|---------------------------|-------------------------------------|
+| seq        | INTEGER  |             | PRIMARY KEY AUTOINCREMENT | Unused on the D side                |
+| envelope   | BLOB     | SCPEnvelope | NOT NULL                  | Serializing a C++ type is dangerous |
+
+
+## Flash database
+
+This contains Flash-related data.
+
+### `channels` table
+
+| Field name    | SQL Type | D type       | Attributes           | Comment                     |
+|---------------|----------|--------------|----------------------|-----------------------------|
+| chan_id       | BLOB     | Hash         | PRIMARY KEY NOT NULL | Binary-serialized hash      |
+| data          | BLOB     | Transaction  | NOT NULL             | Binary-serialized `Channel` |
+| update_signer | BLOB     | UpdateSigner |                      | Doesn't seem to be used     |
+
+
+### `flash_metadata` table
+
+| Field name | SQL Type | D type | Attributes           | Comment                         |
+|------------|----------|--------|----------------------|---------------------------------|
+| meta       | BLOB     | Hash   | PRIMARY KEY NOT NULL | Only ever set to `1` if present |
+| data       | BLOB     |        | NOT NULL             | Binary-serialized `mixin` ?     |
```

Comment:
What does the `?` mean?

Reply:
Means I am not quite sure. CC @AndrejMitrovic .

Reply:
Yes it's binary. I don't know why we should document these? It's going to get outdated anyway.

Reply:
> I don't know why we should document these?

I'm trying to get things under a state in which we  can actually get an overview of the state of a node by looking at the sqlite3 DBs. The first step was documenting everything so we'd know what tables there are and what they are used for. I felt it was a pre-requisite to make the ORM issue feasible.